### PR TITLE
CORTX-31380: Cortx About Information 

### PR DESCRIPTION
**Pull request by abhijit1patil**
Signed-off-by: abhijit1patil <abhijit.1.patil@seagate.com>

# Pull Request
## Problem Statement
-   [CORTX-31380](https://jts.seagate.com/browse/CORTX-31380)

## Design
-   [REST API Specification](https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/1024229377/CORTX+information+REST+API+specification)

-   \[ \] Coding conventions are followed and code is consistent

## Custom Build and Deployment
- [Custom Build Job](https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/6407/)
- [Deployment Job](https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/5754/)
- Custom Build
cortx-docker.colo.seagate.com/seagate/cortx-rgw:2.0.0-6407-custom-ci
cortx-docker.colo.seagate.com/seagate/cortx-data:2.0.0-6407-custom-ci
cortx-docker.colo.seagate.com/seagate/cortx-control:2.0.0-6407-custom-ci

-   \[ \] Unit and System Tests are added
-   \[ \] Test Cases cover Happy Path, Non-Happy Path and Scalability
-   \[ \] Testing was performed with RPM

## Dev Testing
Testcase 0: For public api
image

Testcase 1: with authorization
image

Testcase 2:  get certfication infromation without auth
image

-   \[ \] Interface change (if any) are documented
-   \[ \] Side effects on other features (deployment/upgrade)
-   \[ \] Dependencies on other component(s)

-   \[ \] JIRA number/GitHub Issue added to PR
-   \[ \] PR is self reviewed
-   \[ \] Jira and state/status is updated and JIRA is updated with PR link
-   \[ \] Check if the description is clear and explained

-   \[ \] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: csm/core/blogic/const.py [hunk 1]
```diff
@@ -970,3 +970,7 @@
 SWAGGER_JSON_URL = '/api/doc/swagger.json'
 SWAGGER_UI_STATICS_URL = '/api/static'
 
+# Cortx About Information
+INFORMATION_SERVICE = "information_service"
+ABOUT_INFO_RESOURCES = ['cluster', 'certificate']
+INFORMATION_AUTH_KEY = 'INFORMATION>auth'
```
Comment (pawankumarsrivastava) [inline csm/core/blogic/const.py:976]: Not required for this API.

File: csm/core/controllers/information.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+from csm.core.controllers.view import CsmView, CsmResponse, CsmAuth
+from cortx.utils.log import Log
+from csm.core.blogic import const
+from csm.common.errors import CsmNotFoundError
+
+@CsmAuth.public
+@CsmView._app_routes.view("/api/v2/system/topology")
+class CortxAboutInformationView(CsmView):
```
Comment (pawankumarsrivastava) [inline csm/core/controllers/information.py:23]: CortxAboutInformationView -> CortxInformationView

File: csm/core/controllers/information.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        is_authenticated = False
+        if self.request.session is not None:
+            is_authenticated = True
+        # Call Cortx Information Service
+        response = await self._service.get_cortx_information(is_authenticated)
+        return CsmResponse(response)
+
+@CsmAuth.public
+@CsmView._app_routes.view("/api/v2/system/topology/{resource}")
+class ResourceAboutInformationView(CsmView):
```
Comment (pawankumarsrivastava) [inline csm/core/controllers/information.py:46]: ResourceAboutInformationView -> ResourceInformationView

File: csm/core/controllers/information.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    """
+    def __init__(self, request):
+        super().__init__(request)
+        self._service = self.request.app[const.INFORMATION_SERVICE]
+
+    async def get(self):
+        """GET REST implementation for About Information."""
+        Log.debug("Handling GET request for Cortx About information.")
+        # Check if request is authenticated
+        is_authenticated = False
```
Comment (pawankumarsrivastava) [inline csm/core/controllers/information.py:37]: is_authenticated -> authorized

File: csm/core/controllers/information.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    async def get(self):
+        """GET REST implementation for About Information."""
+        Log.debug("Handling GET request for Cortx About information.")
+        # Read path parameter
+        resource = self.request.match_info[const.ARG_RESOURCE]
+        # Check for valid Resource
+        if resource not in const.ABOUT_INFO_RESOURCES:
+            raise CsmNotFoundError(f"{resource} is not valid")
+        # Check if request is authenticated
+        is_authenticated = False
```
Comment (pawankumarsrivastava) [inline csm/core/controllers/information.py:65]: same as above

File: csm/core/services/information.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    async def get_cortx_information(self, is_authenticated=True, resource=None):
+        """
+        Method to fetch the cortx information
+
+        :param **request_body: Request body kwargs
+        """
+        Log.debug(f"Request body: {resource}")
+        # Call Utils API to get information
+        response = {
+            "is_authenticated" : is_authenticated
```
Comment (pawankumarsrivastava) [inline csm/core/services/information.py:33]: Why is this in response?

**Review by pawankumarsrivastava — CHANGES_REQUESTED** [review]

File: csm/core/services/information.py (hunk 6 above)
Comment (abhijit1patil) [reply csm/core/services/information.py:33]: I want to check whether API works, very well with all the scenario
I just added this for testing.. I will remove it

File: csm/core/blogic/const.py (hunk 1 above)
Comment (abhijit1patil) [reply csm/core/blogic/const.py:976]: Yes this is not required now.. I wil remove it

File: csm/core/controllers/information.py (hunk 2 above)
Comment (abhijit1patil) [reply csm/core/controllers/information.py:23]: updated

File: csm/core/controllers/information.py (hunk 3 above)
Comment (abhijit1patil) [reply csm/core/controllers/information.py:46]: updated

File: csm/core/controllers/information.py (hunk 4 above)
Comment (abhijit1patil) [reply csm/core/controllers/information.py:37]: updated

File: csm/core/controllers/information.py (hunk 5 above)
Comment (abhijit1patil) [reply csm/core/controllers/information.py:65]: updated

**Review by pawankumarsrivastava — APPROVED** [review]

**Pull request merged**